routing.go

```diff
@@ -0,0 +1,64 @@
+package discovery
+
+import (
+	"context"
+	"time"
+
+	cid "github.com/ipfs/go-cid"
+	pstore "github.com/libp2p/go-libp2p-peerstore"
+	routing "github.com/libp2p/go-libp2p-routing"
+	mh "github.com/multiformats/go-multihash"
+)
+
+// RoutingDiscovery is an implementation of discovery using ContentRouting
+type RoutingDiscovery struct {
```

Comment:
Should this not just be:

```go
type RoutingDiscovery struct {
    routing.ContentRouting
}
```
?

Personally, that's how I tend to write adapters like this (and it allows us to avoid the constructor). However, I don't *really* care one way or another.

Reply:
yeah, it's nicer that way; will do.